src/main/java/com/couchbase/connect/kafka/sink/SubDocumentWriter.java

```diff
@@ -96,17 +98,29 @@ public Completable write(final AsyncBucket bucket, final JsonBinaryDocument docu
             }
         }
 
-        return mutation
-                .execute(persistTo, replicateTo)
-                .doOnError(new Action1<Throwable>() {
-                    @Override
-                    public void call(Throwable throwable) {
-                        if(createDocuments && throwable instanceof DocumentDoesNotExistException) {
-                            bucket.insert(JsonDocument.create(document.id())).toBlocking().single();
-                        }
-                    }
-                })
-                .toCompletable();
+         if(createDocuments) {
+             return mutation.execute(persistTo, replicateTo)
```

Comment:
I don't think this is needed, since we have the createDocument() flag on the mutation options which should take care of this in an atomic way, right?

Reply:
i just checked the java client (also in the master branch on github) and i can see a createPath but not a createDocument setting on the mutation options. I might be looking in the wrong class though?

Reply:
sorry its on the builder itself.. as in mutation.createDocument(true)...

Reply:
thanks for the pointer! i'll change it asap